src/strategies/flex_window.py

```diff
@@ -84,170 +86,321 @@ def step(self, event_list=[]):
-
-                        power_vec[ts_idx] = avg_power
-                        safe = sim_vehicle.get_delta_soc() <= 0
-                        if safe:
-                            power_vec[ts_idx+1:] = [0]*(len(timesteps) - ts_idx - 1)
+                    else:
+                        if b.soc < 0 + self.EPS:
+                            # already discharged
                             break
 
-                    if safe:
-                        max_power = power
-                    else:
-                        min_power = power
+                    total_power = (
+                        0
+                        if total_power < b.min_charging_power
+                        else total_power
+                    )
+                    if total_power > 0:
+                        if cur_window:
+                            b.load(
+                                self.interval, (total_power / len(sim_batteries))
+                            )["avg_power"]
+                        else:
+                            b.unload(
+                                self.interval, (total_power / len(sim_batteries))
+                            )["avg_power"]
+            if cur_window:
+                safe = all(
+                    [b.soc >= (1 - self.EPS) for b in sim_batteries])
+            else:
+                safe = all(
+                    [b.soc <= (0 + self.EPS) for b in sim_batteries])
 
-                # apply power
-                if gc.window:
-                    p = power if charged_in_window else gc.cur_max_power - gc.get_current_load()
+            if safe:
+                max_power = total_power
+            else:
+                min_power = total_power
+        # actual charge/ discharge
+        for b_id, battery in self.world_state.batteries.items():
+            if cur_window:
+                avail_power = (
+                    0 if total_power < battery.min_charging_power else total_power
+                )
+                charge = battery.load(self.interval, (avail_power / len(batteries)))["avg_power"]
+                gc.add_load(b_id, charge)
+                timesteps[0]["total_load"] += charge
+#                gc.current_power += charge
+            else:
+                if total_power < 0:
+                    discharge = 0
                 else:
-                    p = 0 if charged_in_window else power
-                p = util.clamp_power(p, vehicle, cs)
-                avg_power = vehicle.battery.load(self.interval, p)['avg_power']
-                charging_stations[cs_id] = gc.add_load(cs_id, avg_power)
-                cs.current_power += avg_power
-
-                for ts_idx, ts_info in enumerate(timesteps):
-                    ts_info["power"] -= power_vec[ts_idx]
+                    discharge = battery.unload(
+                        self.interval, (total_power/len(batteries)),
+                        target_soc=self.DISCHARGE_LIMIT
```

Comment:
stationary batteries are not restricted to a discharge limit, they can be fully emptied. DISCHARGE_LIMIT is strictly for V2G car batteries